docarray/array/storage/memory/find.py

```diff
@@ -53,6 +53,11 @@ def _find(
 
         :return: a list of DocumentArrays containing the closest Document objects for each of the queries in `query`.
         """
+        if 'on' in kwargs.keys():
+            raise ValueError(
+                f'subindices (`on=...`) are not available for this Document Store ({self.__class__.__name__}).'
+            )
```

Comment:
Same, Why is it not available here ? I thought we said it could still be usefull for in memory